reconcile/utils/terrascript_aws_client.py

```diff
@@ -1551,9 +1593,19 @@ def init_populate_specs(
         self.account_resource_specs: dict[str, list[ExternalResourceSpec]] = {}
         self.resource_spec_inventory: ExternalResourceSpecInventory = {}
 
+        # Ensure provider exclusions are fetched
+        if not provider_exclusions_by_provisioner:
+            provider_exclusions_by_provisioner = (
+                queries.get_tf_resources_provider_exclusions_by_provisioner() or []
+            )
+
         for namespace_info in namespaces:
-            specs = get_external_resource_specs(
-                namespace_info, provision_provider=PROVIDER_AWS
+            all_specs = get_external_resource_specs(
+                namespace_info,
+                provision_provider=PROVIDER_AWS,
+            )
+            specs = self._filter_specs_managed_by_erv2(
+                all_specs, provider_exclusions_by_provisioner
```

Comment:
since `provider_exclusions_by_provisioner` will be used for every namespace, to speed up and simplify filter logic, can pre build a dict


```python
provider_exclusions_by_provisioner_name = {
  p["provisioner"]["name"]: p["excludedProviders"]
  for p in provider_exclusions_by_provisioner
}
```

Then filter logic can use it to speed up from O(NM) to O(M), N is length of `provider_exclusions_by_provisioner`, M is length of `excludedProviders`.

```python
            if spec.provider in provider_exclusions_by_provisioner_name.get(spec.provisioner["name"], []):
                raise ProviderExcludedError(spec)
```